plugins/notifications-backend/src/service/router.ts

```diff
@@ -177,6 +181,18 @@ export async function createRouter(
     }
   };
 
+  const validateLink = (link: string) => {
+    const stripLeadingSlash = (s: string) => s.replace(/^\//, '');
+    const ensureTrailingSlash = (s: string) => s.replace(/\/?$/, '/');
+    const url = new URL(
+      stripLeadingSlash(link),
+      ensureTrailingSlash(frontendBaseUrl),
+    );
+    if (url.protocol !== 'https:' && url.protocol !== 'http:') {
```

Comment:
I was thinking whether this is not too restrictive, i.e. for in-app links.
But we can ease the rule later when needed.